Source: Dimitry Ishenko <dimitry.ishenko@gmail.com>
Upstream: yes, issue #935, parts of git commit e3f7713188f1b3205cb96a42724f6a14f64a1a43
Reason: gcc 4.6 with -Werror refuses to compile unused variables;
removing -Werror option causes other problems

--- acpica-unix-20110623/compiler/aslcompiler.l.orig	2011-09-12 13:44:26.130040812 -0400
+++ acpica-unix-20110623/compiler/aslcompiler.l	2011-09-12 13:46:36.116719238 -0400
@@ -550,7 +550,6 @@
     void)
 {
     ASL_FILE_NODE           *Fnode;
-    FILE                    *InputFile = NULL;
 
 
     Fnode = InputStack;
@@ -569,7 +568,6 @@
     /* Update the top-of-stack */
 
     InputStack = Fnode->Next;
-    InputFile = Fnode->File;
 
     /* Reset global line counter and filename */

--- acpica-unix-20110623/os_specific/service_layers/osunixxf.c.orig	2011-09-12 14:18:17.866891275 -0400
+++ acpica-unix-20110623/os_specific/service_layers/osunixxf.c	2011-09-12 14:19:46.230232603 -0400
@@ -349,7 +349,6 @@
     const char              *Fmt,
     va_list                 Args)
 {
-    INT32                   Count = 0;
     UINT8                   Flags;
 
 
@@ -362,7 +361,7 @@
         {
             /* Output file is open, send the output there */
 
-            Count = vfprintf (AcpiGbl_DebugFile, Fmt, Args);
+            vfprintf (AcpiGbl_DebugFile, Fmt, Args);
         }
         else
         {
@@ -374,7 +373,7 @@
 
     if (Flags & ACPI_DB_CONSOLE_OUTPUT)
     {
-        Count = vfprintf (AcpiGbl_OutputFile, Fmt, Args);
+        vfprintf (AcpiGbl_OutputFile, Fmt, Args);
     }
 }
 
